### DISCUSSION TITLE
Make a Blog for MkDocs

### DISCUSSION THREAD

**Discussion by Andre601** [mkdocs/mkdocs#3624] [Ideas]
This was originally posted in an internal discussion with the MkDocs team, but given I left that place for reasons not important here am I posting it here. This may also add the benefit of additional people - mainly end-users - having a chance to share their views on this idea and what they think may or may not be good.

My proposal is to set up a blog within the MkDocs Site to have regular posts in of different topics. This could help in fostering more user-interactions as users could share their feedback through comments (if set up) and/or propose own topics to make.
I can think of the following topics to make posts of:

- In-Depth release-notes, covering the different changes in more detail. This should be limited to updates with substantial changes, therefore patch updates should be ignored (Or alternatively grouped together).
- Posts covering features of MkDocs with more detail, which would be too much to be in a "User Guide" or "Developer Guide" page on their own. This could also cover more niche sections.
- A "Developer Spotlight" or "Project Spotlight" covering a developer or project related to MkDocs respectively. This could give smaller devs or project a spotlight to shine and may even help covering certain questions a user may have. In addition can this help with interaction with the devs by having them communicate with MkDocs when a post is being made.

Of course, there are downsides, namely:

- Time investment. Someone has to invest their time into this to make the posts. I offered myself to do this in the original post, and I stay to that offer.
- Working out a schedule. Posts should be made regularely for a level of consistency, so a schedule should be worked out here. This also means that topics for future posts should also be worked out, so that it can be prepared ahead of time.
- Setting the blog up. If we stick to having the block in MkDocs - which I think would be best for ease of maintenance - would it need work. No matter if a plugin is used (recommended) or it's done manually, making pages will require work. Using a plugin (i.e. Material for MkDocs' blog plugin which seems to be the best choice here) would certainly require template work (unless the theme it comes in would be used).

I want to stress that this is not a necessary thing, nor that it has any level of importance. I'm fine with keeping this on the backburner until the Project itself has reached a proper setup and structure to maintain.
My main focus with this discussion is to gather user feedback on this to see if people may consider this a good addition in the future for MkDocs.

**Comment by pawamoy** [comment]
As said internally, I like the idea. We're not ready yet to setup such a blog, but as soon as we are I'm interested in participating.
In the meantime what I can suggest is to publish such blog posts on your own website/blog, and when the time comes and we have a blog ready, suggest to move/copy them into MkDocs.

Also, even if you commit to it (thanks!), we need to have many different authors (it would be weird to publish the posts of a single  person all the time). Let see how much interest this discussion creates, and tag back all those who participated when we're ready to tackle the blog idea :slightly_smiling_face: If there's not a lot of interest, we can also consider just doing project spotlights in a semi-automated way (the catalog can help with that).

**Reply by Andre601** [reply]
> In the meantime what I can suggest is to publish such blog posts on your own website/blog, and when the time comes and we have a blog ready, suggest to move/copy them into MkDocs.

Already have some posts related to MkDocs, mainly...

- [How to make a Comment System using Mastodon](https://andre601.ch/blog/2023/07-23-setup-mastodon-comment-system/) and
- [How to publish MkDocs pages to Codeberg Pages](https://andre601.ch/blog/2023/11-05-using-codeberg-pages/)

The most difficult part for me is to find further topics a User may be interested. And I do lack the cloud to gather proper feedback on any possible ideas...

**Reply by Andre601** [reply]
The above message was btw a small hint that I'm always open for suggestions on new topics to cover, regarding MkDocs, so if you have some, let me know ;)

**Comment by vedranmiletic** [comment]
@Andre601

> Posts covering features of MkDocs with more detail, which would be too much to be in a "User Guide" or "Developer Guide" page on their own. This could also cover more niche sections.

@pawamoy

> In the meantime what I can suggest is to publish such blog posts on your own website/blog, and when the time comes and we have a blog ready, suggest to move/copy them into MkDocs.

I am not a contributor, just an enthusiastic user of (Material for) MkDocs and I [already](https://group.miletic.net/en/blog/2021-08-16-markdown-vs-restructuredtext-for-teaching-materials/) [blogged](https://group.miletic.net/en/blog/2022-11-01-publishing-material-for-mkdocs-website-to-github-pages-using-custom-actions-workflow/) about it in the past. There is another post in the pipeline about migration from WordPress to (Material for) MkDocs, based on the experience with [linuxhr/hulk-ri](https://github.com/linuxhr/hulk-ri). I would be glad to contribute any of these to the official blog.
